bindings/go/scip/canonicalize.go

```diff
@@ -10,13 +10,29 @@ func CanonicalizeDocument(document *Document) *Document {
 // CanonicalizeOccurrences deterministically re-orders the fields of the given occurrence slice.
 func CanonicalizeOccurrences(occurrences []*Occurrence) []*Occurrence {
 	canonicalized := make([]*Occurrence, 0, len(occurrences))
-	for _, occurrence := range FlattenOccurrences(occurrences) {
+	for _, occurrence := range FlattenOccurrences(RemoveIllegalOccurrences(occurrences)) {
 		canonicalized = append(canonicalized, CanonicalizeOccurrence(occurrence))
 	}
 
 	return SortOccurrences(canonicalized)
 }
 
+// RemoveIllegalOccurrences removes all occurrences that do not include a range. This is
+// emitted by some indexers and will silently crash a downstream process, including further
+// canonicalization, when trying to convert an empty slice into a valid range.
+func RemoveIllegalOccurrences(occurrences []*Occurrence) []*Occurrence {
+	filtered := occurrences[:0]
+	for _, occurrence := range occurrences {
+		if len(occurrence.Range) == 0 {
+			continue
+		}
```

Comment:
We should be stricter here, I think? Only length 3 and length 4 occurrences are actually valid per the SCIP schema.

Reply:
[9167b70](https://github.com/sourcegraph/scip/pull/180/commits/9167b70dab12bf4ae9ee3eeabb4b5ebedd50345a)